pkg/eventsapi/generic_event.go

```diff
@@ -0,0 +1,48 @@
+package eventsapi
+
+import (
+	"encoding/json"
+)
+
+type GenericEvent struct {
+	EventVersion EventVersion
+	EventData    map[string]interface{}
+}
+
+func (ge GenericEvent) Validate() error {
+	if ge.EventVersion == EventVersion1 {
+		event := ge.getSpecificV1EventStruct()
+		return event.Validate()
+	} else {
+		event := ge.getSpecificV2EventStruct()
+		return event.Validate()
+	}
+}
+
+func (ge GenericEvent) GetRoutingKey() string {
+	if ge.EventVersion == EventVersion1 {
+		event := ge.getSpecificV1EventStruct()
+		return event.GetRoutingKey()
+	} else {
+		event := ge.getSpecificV2EventStruct()
+		return event.GetRoutingKey()
+	}
+}
+
+func (ge GenericEvent) Version() EventVersion {
+	return ge.EventVersion
+}
+
+func (ge GenericEvent) getSpecificV1EventStruct() EventV1 {
+	jsonV1Event, _ := json.Marshal(ge.EventData)
```

Comment:
Could we potentially avoid having to `Marshal` / `Unmarshal` if we constructed this object slightly differently (maybe by storing `json.RawMessage` or something?)

Reply:
Yeah, this make sense, fixed here: 2abcdd4